Informative note about role of id attribute

ianbjacobs opened the pull request:
closes #911 

The following tasks have been completed:

 * [x] Confirmed there are no ReSpec errors/warnings.
 * [ ] Has undergone security/privacy review (link)


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/payment-request/pull/913.html" title="Last updated on May 7, 2020, 12:38 PM UTC (057492c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/payment-request/913/f868c93...057492c.html" title="Last updated on May 7, 2020, 12:38 PM UTC (057492c)">Diff</a>

ianbjacobs committed:
Informative note per
https://github.com/w3c/payment-request/issues/911

marcoscaceres commented on index.html:
```diff
@@ -917,6 +917,14 @@ <h2>
+        <div class="note">
+          <p>
+            For auditing and reconciliation purposes, a merchant can assign a
+            unique identifier for each transaction with the <a>id</a>
```
```suggestion
            unique identifier for each transaction with the {{PaymentDetailsInit/id}}
```

marcoscaceres commented:
Seems a little bit redundant to add this, but 🤷‍♂️.

ianbjacobs committed:
simplified note re: id

ianbjacobs commented:
@marcoscaceres, I simplified the note to the one sentence about the purpose of the attribute (which is not stated elsewhere as far as I can tell). I removed the bit about browser-generated ID when not provided by the merchant; that is covered in the algorithm.

marcoscaceres reviewed (approved):
Lgtm

ianbjacobs commented:
(Wondering why checks haven't completed...)

marcoscaceres commented:
Used my super powers.
